lib/puppet/type/mount.rb

```diff
@@ -293,5 +293,8 @@ def value(name)
       dependencies[0..-2]
     end
 
+    # Autorequire the mount point's file resource
+    autorequire(:file) { Pathname.new(@parameters[:name].value) }
```

Comment:
We now have a autobefore method. The file type is pretty core, where-as the mount type isn't as central and may later be moved to a separate module (that would generally be part of the standard distribution, but could be removed). How about moving the dependency logic for mount <-> file to an autobefore in the mount type, so the File doesn't have to know about mounts?